src/Tag/Tag.tsx

```diff
@@ -4,39 +4,63 @@ import styled from 'styled-components'
 import { Text } from '../Text'
 import { Color, theme } from '../theme'
 
-export type TagProps = {
+export type DefaultProps = {
   label: string
   className?: string
   color: Color
-  bgColor: Color
-  borderColor: Color
   typo?: string
 }
 
+type GradientProps =
+  | {
+      // if bgGradient === false
+      bgColor?: Color
+      borderColor?: Color
+      bgGradient?: false
+    }
+  | {
+      // if bgGradient === true
+      bgColor: Color
+      borderColor: Color
+      bgGradient: true
```

Comment:
I don't think this works. It should be
```
| {
      // no gradient
      bgColor: Color
      borderColor: Color
      bgGradient?: boolean
    }
  | {
      // with gradient
      bgColor?: Color
      borderColor?: Color
      bgGradient: boolean
}
```

We are trying to define which props are mandatory and which one are optional based on gradient vs solid colours